zebra-consensus/src/checkpoint/list.rs

```diff
@@ -55,16 +54,33 @@ const TESTNET_CHECKPOINTS: &str = include_str!("test-checkpoints.txt");
 #[derive(Clone, Debug, Eq, Hash, PartialEq)]
 pub struct CheckpointList(BTreeMap<block::Height, block::Hash>);
 
-impl FromStr for CheckpointList {
-    type Err = BoxError;
+impl CheckpointList {
+    /// Returns the hard-coded checkpoint list for `network`.
+    pub fn new(network: Network) -> Self {
+        // parse calls CheckpointList::from_list
+        let checkpoint_list: CheckpointList = match network {
+            Network::Mainnet => Self::from_str(MAINNET_CHECKPOINTS, network)
+                .expect("Hard-coded Mainnet checkpoint list parses and validates"),
+            Network::Testnet(_) => Self::from_str(TESTNET_CHECKPOINTS, network)
+                .expect("Hard-coded Testnet checkpoint list parses and validates"),
+        };
+
+        match checkpoint_list.hash(block::Height(0)) {
+            Some(hash) if hash == genesis_hash(network) => checkpoint_list,
+            Some(_) => {
+                panic!("The hard-coded genesis checkpoint does not match the network genesis hash")
+            }
+            None => unreachable!("Parser should have checked for a missing genesis checkpoint"),
+        }
+    }
```

Comment:
This is only correct if `network == Network::new_testnet()`. If it's not, then the list should consist of only the checkpoints supplied by the custom network. Currently that's a single item list containing the genesis block hash.

If we implement it that way, then there's no need to check the genesis checkpoint in production code, and that check can be moved to a unit test.